test/differentials.jl

```diff
@@ -31,16 +31,25 @@
     @testset "One" begin
         o = One()
         @test extern(o) === true
-        @test add_one(o, o) == 2
-        @test add_one(o, 1) == 2
-        @test add_one(1, o) == 2
-        @test mul_one(o, o) == o
-        @test mul_one(o, 1) == 1
-        @test mul_one(1, o) == 1
+        @test +(o, o) == 2
+        @test +(o, 1) == 2
+        @test +(1, o) == 2
+        @test *(o, o) == o
+        @test *(o, 1) == 1
+        @test *(1, o) == 1
         for x in o
             @test x === o
         end
         @test broadcastable(o) isa Ref{One}
         @test conj(o) == o
     end
+
+
+    @testset "No ambiguities in $f" for f in (+, *)
```

Comment:
Is this testset a bit broad? Are we not testings literally _all_ methods of + and *, including those defined in Base? Is this an issue?

Reply:
`Test.detect_ambiguities` tests all methods in the module.
Which is much broader.

Cutting this down is fiddly.

Should probably make it just one test though. As it currently massive inflates our test count